### PR TITLE
 amp-shadow: <noscript> gets parsed as markup when parsing HTML strings!

### LINKED ISSUE #12132: amp-shadow does not support amp-img media query
When using amp-shadow and amp-img with media query such:
`<amp-img ... media="(min-width: 768px)"..>`
`<amp-img ... media="(max-width: 767px)"..>`
the images are requested twice, meaning that the media attribute is not correctly used.
If you load [amp-by-example-staging](https://amp-by-example-staging.appspot.com/polymer/detail/introduction/hello_world/) and open the Google console you should see the images using media loaded for both the resolution.
@cvializ FYI

### PR DESCRIPTION
**Pull request by aghassemi**
Closes #12132

`noscript` parses as markup in both `parseFromString` and `createHTMLDocument`

Apparently WAI: spec: https://w3c.github.io/DOM-Parsing/#the-domparser-interface

/to @dvoytenko @cvializ

### CODE REVIEW

File: test/functional/test-shadow-embed.js [hunk 1]
```diff
@@ -478,6 +478,24 @@ describes.sandboxed('shadow-embed', {}, () => {
         });
       });
     });
+
+    it('should not parse noscript as markup', () => {
```
Comment (dvoytenko) [inline test/functional/test-shadow-embed.js]: Can we also have a test that splits `<noscript>` and `</noscript>` in multiple writes? My concern is that I'm not sure what it'd do while parsing int he middle of `<noscript>` when the node is removed.

File: src/shadow-embed.js [hunk 2]
```diff
@@ -692,3 +696,20 @@ export class ShadowDomWriterBulk {
...
+ * @param {!Element} parent
+ *
+ * According to the spec (https://w3c.github.io/DOM-Parsing/#the-domparser-interface),
+ * with `DOMParser().parseFromString`, contents of `noscript` get parsed as markup,
+ * so we need to remove them manually.
+ * Why? ¯\_(ツ)_/¯
+ * `createHTMLDocument()` seems to behave the same way.
+ */
+function removeNoScriptElements(parent) {
+  const noscriptElements = toArray(childElementsByTag(parent, 'noscript'));
```
Comment (cvializ) [inline src/shadow-embed.js]: Is there a particular reason you use `toArray` and `forEach` over `iterateCursor`?

File: test/functional/test-shadow-embed.js (hunk 1 above)
Comment (aghassemi) [reply test/functional/test-shadow-embed.js]: Added test, seems to work fine.

File: src/shadow-embed.js (hunk 2 above)
Comment (aghassemi) [reply src/shadow-embed.js]: didn't know about `iterateCursor`, neat!

**Comment by aghassemi** [comment]
@dvoytenko Sorry took me a bit to get back to this, PTAL.

**Pull request merged**